elftools/elf/elffile.py

```diff
@@ -94,15 +95,25 @@ def load_from_path(cls, path):
         ELFFile from it, setting up a correct stream_loader relative to the
         original file.
         """
-        base_directory = os.path.dirname(path)
-        def loader(elf_path):
-            # FIXME: use actual path instead of str/bytes
-            if not os.path.isabs(elf_path):
-                elf_path = os.path.join(base_directory,
-                                        elf_path)
-            return open(elf_path, 'rb')
         stream = open(path, 'rb')
-        return ELFFile(stream, loader)
+        return ELFFile(stream, ELFFile.make_relative_loader(path))
+    
+    @staticmethod
+    def make_relative_loader(base_path):
+        """ Return a function that takes a potentially relative path,
+            resolves it against base_path (bytes or str), and opens a file at that.
+
+            ELFFile uses functions like that for resolving DWARF links.
+        """
+        if isinstance(base_path, str):
+            base_path = base_path.encode('UTF-8') # resolver takes a bytes path
+        base_directory = os.path.dirname(base_path)
+        def loader(rel_path):
+            # FIXME: use actual path instead of str/bytes
```

Comment:
I'm not sure what this FIXME means, can you clarify?

Reply:
The FIXME goes back to @rdunklau and #426. I presume it was meant to say that paths should be passed around as `pathlib.Path` objects.

Reply:
I'll say just remove the FIXME for now. It's not helpful, IMHO. If someone wants to refactor to use `pathlib.Path`, this will have to be done in more places than just here, anyway